cli/server_test.go

```diff
@@ -1836,7 +1836,7 @@ func TestConnectToPostgres(t *testing.T) {
 	dbURL, err := dbtestutil.Open(t)
 	require.NoError(t, err)
 
-	sqlDB, err := cli.ConnectToPostgres(ctx, log, "postgres", dbURL)
+	sqlDB, err := cli.ConnectToPostgres(ctx, log, true, "postgres", dbURL)
```

Comment:
If we do the control flag approach, we should add a separate test here to ensure that no migrations are run if `migrate=false`.